### PR TITLE
[Cache][Config][Console][DependencyInjection][FrameworkBundle] Remove unused code and useless casts

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/CacheWarmer/RouterCacheWarmer.php [hunk 1]
```diff
@@ -43,7 +43,7 @@ public function warmUp(string $cacheDir, ?string $buildDir = null): array
         $router = $this->container->get('router');
 
         if ($router instanceof WarmableInterface) {
-            return (array) $router->warmUp($cacheDir, $buildDir);
+            return $router->warmUp($cacheDir, $buildDir);
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/CacheWarmer/RouterCacheWarmer.php]: The test failure for the low deps job looks related to this change.

Comment (alexandre-daubois) [reply src/Symfony/Bundle/FrameworkBundle/CacheWarmer/RouterCacheWarmer.php]: Actually the same problem as #57775. I added [the same line](https://github.com/symfony/symfony/pull/57814/files#diff-f9f5e8b655786315f3693fd571bc1d5f07e419805bfce50c74b7b1a82804c332R27) in this PR, let's see which gets merged first :slightly_smiling_face:

Comment (alexandre-daubois) [reply src/Symfony/Bundle/FrameworkBundle/CacheWarmer/RouterCacheWarmer.php]: The other one got merged, I rebased.

File: src/Symfony/Component/Console/Descriptor/ReStructuredTextDescriptor.php [hunk 2]
```diff
@@ -92,7 +92,7 @@ protected function describeInputOption(InputOption $option, array $options = [])
     protected function describeInputDefinition(InputDefinition $definition, array $options = []): void
     {
         if ($showArguments = ((bool) $definition->getArguments())) {
-            $this->write("Arguments\n".str_repeat($this->subsubsectionChar, 9))."\n\n";
```
Comment (xabbuh) [inline src/Symfony/Component/Console/Descriptor/ReStructuredTextDescriptor.php:95]: Why this change?

Comment (alexandre-daubois) [reply src/Symfony/Component/Console/Descriptor/ReStructuredTextDescriptor.php:95]: The double newline is outside the `write()` call. Seems too weird to be intentional, I guess it has never done anything then

**Review by xabbuh — APPROVED** [review]

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 3]
```diff
@@ -1020,7 +1020,7 @@ private function addInlineReference(string $id, Definition $definition, string $
             return $code;
         }
 
-        $code .= \sprintf(<<<'EOTXT'
+        return $code . \sprintf(<<<'EOTXT'
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]:
```suggestion
        return $code.\sprintf(<<<'EOTXT'
```

**Comment by nicolas-grekas** [comment]
Thank you @alexandre-daubois.
